platform/platform_tests/distance_tests.cpp

```diff
@@ -238,97 +256,99 @@ UNIT_TEST(Distance_FormattedDistance)
-    {Distance(287'386, Units::Meters),     287,  Units::Kilometers, "287",  "287 km"},
+    {Distance(999,     Units::Meters),     1.0,   Units::Kilometers, "1.0",    "1.0\u202Fkm"},
+    {Distance(1000,    Units::Meters),     1.0,   Units::Kilometers, "1.0",    "1.0\u202Fkm"},
+    {Distance(1001,    Units::Meters),     1.0,   Units::Kilometers, "1.0",    "1.0\u202Fkm"},
+    {Distance(1100,    Units::Meters),     1.1,   Units::Kilometers, "1.1",    "1.1\u202Fkm"},
+    {Distance(1140,    Units::Meters),     1.1,   Units::Kilometers, "1.1",    "1.1\u202Fkm"},
+    {Distance(1151,    Units::Meters),     1.2,   Units::Kilometers, "1.2",    "1.2\u202Fkm"},
+    {Distance(1500,    Units::Meters),     1.5,   Units::Kilometers, "1.5",    "1.5\u202Fkm"},
+    {Distance(1549.9,  Units::Meters),     1.5,   Units::Kilometers, "1.5",    "1.5\u202Fkm"},
+    {Distance(1550,    Units::Meters),     1.6,   Units::Kilometers, "1.6",    "1.6\u202Fkm"},
+    {Distance(1551,    Units::Meters),     1.6,   Units::Kilometers, "1.6",    "1.6\u202Fkm"},
+    {Distance(9949,    Units::Meters),     9.9,   Units::Kilometers, "9.9",    "9.9\u202Fkm"},
+    {Distance(9992,    Units::Meters),     10,    Units::Kilometers, "10",     "10\u202Fkm"},
+    {Distance(10000,   Units::Meters),     10,    Units::Kilometers, "10",     "10\u202Fkm"},
+    {Distance(10499.9, Units::Meters),     10,    Units::Kilometers, "10",     "10\u202Fkm"},
+    {Distance(10501,   Units::Meters),     11,    Units::Kilometers, "11",     "11\u202Fkm"},
+    {Distance(101'001, Units::Meters),     101,   Units::Kilometers, "101",    "101\u202Fkm"},
+    {Distance(101'999, Units::Meters),     102,   Units::Kilometers, "102",    "102\u202Fkm"},
+    {Distance(287'386, Units::Meters),     287,   Units::Kilometers, "287",    "287\u202Fkm"},
 
     // From Feet to Miles
-    {Distance(999,     Units::Feet),       0.2,  Units::Miles,      "0.2",  "0.2 mi"},
-    {Distance(1000,    Units::Feet),       0.2,  Units::Miles,      "0.2",  "0.2 mi"},
-    {Distance(1150,    Units::Feet),       0.2,  Units::Miles,      "0.2",  "0.2 mi"},
-    {Distance(5280,    Units::Feet),       1.0,  Units::Miles,      "1.0",  "1.0 mi"},
-    {Distance(7920,    Units::Feet),       1.5,  Units::Miles,      "1.5",  "1.5 mi"},
-    {Distance(10560,   Units::Feet),       2.0,  Units::Miles,      "2.0",  "2.0 mi"},
-    {Distance(100'000, Units::Feet),       19,   Units::Miles,      "19",   "19 mi"},
-    {Distance(285'120, Units::Feet),       54,   Units::Miles,      "54",   "54 mi"},
-    {Distance(633'547, Units::Feet),       120,  Units::Miles,      "120",  "120 mi"},
-    {Distance(633'600, Units::Feet),       120,  Units::Miles,      "120",  "120 mi"},
-    {Distance(633'653, Units::Feet),       120,  Units::Miles,      "120",  "120 mi"},
-    {Distance(999'999, Units::Feet),       189,  Units::Miles,      "189",  "189 mi"},
+    {Distance(999,     Units::Feet),       0.2,   Units::Miles,      "0.2",    "0.2\u202Fmi"},
```

Comment:
Let's move `\u202F` to a const variable at the beginning of the file:
`char const kNonBreakingSpace = '\u202F'`

Reply:
constexpr is even better.

Reply:
As the non-breaking space `\u202F` is used in different source files, I've included its definition in the file `locale.hpp`.

I cannot get it working as `char`, so I've declared it as `std::string`.